source/material_model/visco_plastic.cc

```diff
@@ -493,84 +383,17 @@ namespace aspect
-    {
-      // We need the velocity gradient for the finite strain (they are not included in material model inputs),
-      // so we get them from the finite element.
-
-      std::vector<Point<dim> > quadrature_positions(in.position.size());
-      for (unsigned int i=0; i < in.position.size(); ++i)
-        quadrature_positions[i] = this->get_mapping().transform_real_to_unit_cell(in.current_cell, in.position[i]);
-
-      FEValues<dim> fe_values (this->get_mapping(),
-                               this->get_fe(),
-                               Quadrature<dim>(quadrature_positions),
-                               update_gradients);
-
-      std::vector<Tensor<2,dim> > velocity_gradients (quadrature_positions.size(), Tensor<2,dim>());
-
-      fe_values.reinit (in.current_cell);
-      fe_values[this->introspection().extractors.velocities].get_function_gradients (this->get_solution(),
-                                                                                     velocity_gradients);
-
-      // Assign the strain components to the compositional fields reaction terms.
-      // If there are too many fields, we simply fill only the first fields with the
-      // existing strain tensor components.
-      for (unsigned int q=0; q < in.position.size(); ++q)
-        {
-          // Convert the compositional fields into the tensor quantity they represent.
-          Tensor<2,dim> strain;
-          const unsigned int n_first = this->introspection().compositional_index_for_name("s11");
-          for (unsigned int i = n_first; i < n_first + Tensor<2,dim>::n_independent_components ; ++i)
-            {
-              strain[Tensor<2,dim>::unrolled_to_component_indices(i)] = in.composition[q][i];
-            }
-
-          // Compute the strain accumulated in this timestep.
-          const Tensor<2,dim> strain_increment = this->get_timestep() * (velocity_gradients[q] * strain);
-
-          // Output the strain increment component-wise to its respective compositional field's reaction terms.
-          for (unsigned int i = n_first; i < n_first + Tensor<2,dim>::n_independent_components ; ++i)
-            {
-              out.reaction_terms[q][i] = strain_increment[Tensor<2,dim>::unrolled_to_component_indices(i)];
-            }
-        }
-    }
-
-
-
     template <int dim>
     ComponentMask
     ViscoPlastic<dim>::
     get_volumetric_composition_mask() const
     {
-      // Store which components to exclude during volume fraction computation.
-      ComponentMask composition_mask(this->n_compositional_fields(),true);
-      if (weakening_mechanism != none)
-        {
-          if (weakening_mechanism == plastic_weakening_with_plastic_strain_only || weakening_mechanism == plastic_weakening_with_plastic_strain_and_viscous_weakening_with_viscous_strain)
-            composition_mask.set(this->introspection().compositional_index_for_name("plastic_strain"),false);
+      // Return composition mask for strain fields
```

Comment:
The comments in this function are not really clear to me. I liked the previous version: `Store which components to exclude during volume fraction computation.`, because it explains what this composition mask *is used for*, not what it is. Please just revert to the old comment.